Tools/TestWebKitAPI/Tests/WebKitCocoa/RestoreLocalStorage.mm

```diff
@@ -119,8 +174,46 @@ async function setItem()
 </script>
 )TESTRESOURCE"_s;
 
-static void testFetchLocalStorageThirdPartyIFrame(RetainPtr<WKWebsiteDataStore> websiteDataStore)
+static NSString *getMainFrameString = @"<script> \
+    function postResult(event) \
+    { \
+        window.webkit.messageHandlers.testHandler.postMessage(event.data); \
+    } \
+    addEventListener('message', postResult, false); \
+    </script> \
+    <iframe src='https://127.0.0.1:9091/'>";
+
+static constexpr auto getFrameBytes = R"TESTRESOURCE(
```

Comment:
I don't think you need to separate `getFrameBytes` and `setFrameBytes`. Just have one `frameBytes`, which invokes `getItem()` first, if item exists, post message with "found item [value]" and return; if not exists, `setItem()` and post message "set item".
You could tell whether item exists by the message.

Reply:
That's a good point. Fixed!